docs(design): add description for construction_order (#129)
tier4/CARET_doc: docs/design/configuration/architecture_file.md

        subscription_topic_name: /pong
        publisher_topic_name: /ping
```

## Callback identification

`callback_name` is a name used by users to identify a callback.

> CARET reads trace data and loads callback information.
> Callback addresses can be obtained from trace data,
> but callback_name cannot be obtained because ROS does not have the ability to assign callback_names.

Reviewer: ```suggestion
CARET helps users to give a name to a callback, but it is not directly associated with its address due to the reason as explained above.
```

> After loading information, CARET binds the `callback_name` and callback address by following items.

Reviewer: In order to tackle the issue, CARET associates a name with an address of callback with using combination of following data.

Reviewer: ```suggestion
In order to tackle the issue, CARET associates a name with an address of callback with using combination of following data.
```

> - node_name
> - callback_type
> - period_ns / topic_name
> - symbol
> - construction_order

Reviewer: ```suggestion
- `node_name`
- `callback_type`
- `period_ns` / `topic_name`
- `symbol`
- `construction_order`
```

By using this information to match `callback_name` and callback address,

> `callback_name` will always refer to identical callbacks, regardless of the number of recordings.

Reviewer: ```suggestion
each `callback_name` will always be unique if there are multiple callbacks instantiated from a same function definition.
```

Author: I added this note because there was a part that was difficult to understand.
Please confirm.

d25ef82

        subscription_topic_name: /pong
        publisher_topic_name: /ping
```

## Callback identification

> It's convenience for users to give a name to a callback function for its identification. However, in the context of ROS 2, only an address is given to a callback.

Reviewer: ```suggestion
It's convenient for users to give a name to a callback function for its identification. However, in the context of ROS 2, only an address is given to a callback.
```

Addresses change with each launch of an application.
This makes it difficult to handle callbacks by address when evaluating performance of them.
For example, if you want to compare the execution time of a particular callback for each launch, you have to find address to select target callbacks.

CARET helps users to give a name to a callback, but it is not directly associated with its address due to the reason as explained above.

In order to tackle the issue, CARET associates a name with an address of callback with using combination of following data.

- `node_name`
- `callback_type`
- `period_ns` / `topic_name`
- `symbol`
- `construction_order`

By using this information to match `callback_name` and callback address,

> each `callback_name` will always refer to identical callbacks without regarding to callback address.

Reviewer: ```suggestion
each `callback_name` will always refer to identical callbacks without being aware of callback address.
```
